src/sparseml/pytorch/sparsification/quantization/helpers.py

```diff
@@ -433,68 +708,92 @@ def remove_activation_qat_by_layer_name(module: Module, layer_class_names: List[
-        will use asymmetric quantization with any zero point. Default is False
+        quantization range with a pre-specified zero point
+        (0 if activation_dtype=torch.qint8, 128 if activation_dtype=torch.quint8).
+        Default is False.
     :param symmetric_weights: if True, weights will have a symmetric
-        INT8 quantization range with zero point set to 0. Otherwise activations
-        will use asymmetric quantization with any zero point. Default is True
+        quantization range with a pre-specified zero point
+        (0 if weight_dtype=torch.qint8, 128 if weight_dtype=torch.quint8).
+        Default is True.
     :param reduce_range: if True, the quantization range will be reduced by one bit.
-        This may prevent overflow issues with model execution on certain hardware
-        Default is False
+        This may prevent overflow issues with model execution on certain hardware.
+        Default is False.
     :param activation_qconfig_kwargs: Additional kwargs for quantization of
-            activations.
+        activations.
     :param weight_qconfig_kwargs: Additional kwargs for quantization of
-            weights.
+        weights.
+    :param activation_dtype: quantized activation data type. Default is torch.quint8.
+    :param weight_dtype: quantized weights data type. Default is torch.qint8.
+    :param activation_bits: number of bits for activations. Default is 8.
+    :param weight_bits: number of bits for weights. Default is 8.
     :return: A QAT fake quantization config for symmetric weight quantization and
         asymmetric activation quantization.  The difference between this and
         torch.quantization.default_qat_qconfig is that the activation observer
         will not have reduce_range enabled.
     """
-    activation_qscheme = (
-        torch.per_tensor_symmetric if symmetric_activations else torch.per_tensor_affine
-    )
-    activation_observer_kwargs = dict(
-        observer=torch_quantization.MovingAverageMinMaxObserver,
-        quant_min=0,
-        quant_max=255,
-        dtype=torch.quint8,
-        qscheme=activation_qscheme,
-        reduce_range=reduce_range,
-    )
-    activation_observer_kwargs.update(activation_qconfig_kwargs or {})
-    activation_observer = torch_quantization.FakeQuantize.with_args(
-        **activation_observer_kwargs,
-    )
-    weight_qscheme = (
-        torch.per_tensor_symmetric if symmetric_weights else torch.per_tensor_affine
-    )
-    weight_observer_kwargs = dict(
-        observer=torch_quantization.MovingAverageMinMaxObserver,
-        quant_min=-128,
-        quant_max=127,
-        dtype=torch.qint8,
-        qscheme=weight_qscheme,
-        reduce_range=reduce_range,
+    activation_observer = get_observer(
+        symmetric_activations,
+        activation_dtype,
+        activation_bits,
+        reduce_range,
+        activation_qconfig_kwargs,
     )
+    if symmetric_weights is None:
```

Comment:
why not just default to True?

Reply:
I agree that setting it to True makes sense here.